forum/profiles/models.py

```diff
@@ -111,3 +114,39 @@ def __str__(self):
             f"email={self.email}"
             f")"
         )
+
+
+class ViewedStratups(models.Model):
+    """
+    Model to track startup views by investors.
+
+    Fields:
+        startup (ForeignKey): Reference to the StartupProfile model representing the viewed startup.
+        investor (ForeignKey): Reference to the InvestorProfile model.
+        viewed_at (DateTimeField): The timestamp when the startup was viewed.
+    """
+    startup = models.ForeignKey(StartupProfile, on_delete=models.CASCADE)
+    investor = models.ForeignKey(InvestorProfile, on_delete=models.CASCADE, db_index=True)
+    viewed_at = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=['investor', 'startup'],
+                name='one_view_per_day',
+                condition=Q(viewed_at__date=TruncDate('viewed_at'))
+            )
+        ]
+        ordering = ["-viewed_at"]
```

Comment:
Consider adding a composite index on `investor` and `viewed_at` to optimize queries that filter by these fields.
```
indexes = [
    models.Index(fields=['investor', 'viewed_at'], name='investor_viewed_at_idx'),
]
```